sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,267 @@
+            .forPort(3500)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    APP2_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("app2")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("app2-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withCustomHttpPort(3501)
+        .withCustomGrpcPort(50002)
+        .withAppChannelAddress("main-workflow-sidecar:3500")
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1",
+            Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withLogConsumer(outputFrame -> System.out.println("APP2: " + outputFrame.getUtf8String()))
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
+            .forPort(3501)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    APP3_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("app3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("app3-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withCustomHttpPort(3502)
+        .withCustomGrpcPort(50003)
+        .withAppChannelAddress("main-workflow-sidecar:3500")
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1",
+            Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withLogConsumer(outputFrame -> System.out.println("APP3: " + outputFrame.getUtf8String()))
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
+            .forPort(3502)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    // Start the Dapr containers
+    MAIN_WORKFLOW_CONTAINER.start();
+    APP2_CONTAINER.start();
+    APP3_CONTAINER.start();
+
+    // Start crossapp worker (connects to MAIN_WORKFLOW_CONTAINER)
+    crossappWorker = new GenericContainer<>("openjdk:17-jdk-slim")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/test-classes"), "/app/classes")
+        .withCopyFileToContainer(MountableFile.forHostPath("target/dependency"), "/app/libs")
+        .withWorkingDirectory("/app")
+        .withCommand("java", "-cp", "/app/classes:/app/libs/*", 
+                     "-Ddapr.app.id=crossapp-worker",
+                     "-Ddapr.grpc.endpoint=main-workflow-sidecar:50001",
+                     "-Ddapr.http.endpoint=main-workflow-sidecar:3500",
```

Comment:
@cicoyle this should use Testcontainers mapped ports instead of hardcoded ports.

Reply:
Actually, this is using internal container communication within the Docker network, not host communication. The worker containers are all running in the same Docker network and communicate directly with each other using network aliases like `main-workflow-sidecar:50001`. 

For internal container-to-container communication, I believe we need to use the internal network names and ports, not mapped host ports. The mapped ports are only needed when communicating from the host (test code) to containers.

Reply:
@cicoyle I think that now is good, because at least testcontainers is starting the containers in random ports.